backend/types.py

```diff
@@ -121,6 +122,18 @@ class ModelProviderConfig(BaseModel):
     reranking_model_ids: List[str]
     api_key_env_var: str
     base_url: Optional[str] = None
+    headers: Optional[List] = None
+
+    @property
+    def get_headers(self) -> Dict[str, str]:
+        formatted_headers = {}
+        for header in self.headers:
+            for key, value in header.items():
+                if isinstance(value, dict):
+                    formatted_headers[key] = json.dumps(value)
+                else:
+                    formatted_headers[key] = value
+        return formatted_headers
```

Comment:
No need for this complexity in my opinion, let user's format in the yaml itself

Reply:
gateway requires dict to be json.dumps()